bot/cogs/grades_data/grades_cog.py

```diff
@@ -277,5 +341,42 @@ async def grades(self, ctx, course, year=MIN_YEAR):
 
         await ctx.send(embed=embed)
 
+    @commands.command()
+    async def prof(self, ctx, firstName, lastName, detailed = bool('yes')):
+        # Should not need a min year as professors hardly change a significant amount to be noteworthy (Exception: SP 2020 -- we ignore those dark times)
+        """
+        Attempts to give more information about professor's grade distribution @ Clemson.
+
+        USE:
+
+        prof <first name> <last name> <detailed (Optional)>
+        EX: !prof Brian Dean                <-- will print all letter distributions
+        !prof Brian Dean blah blah blah     <-- will print only pass/fail/withdrawal
+
+        DISCLAIMER:
+        Not every professor listed will be at Clemson, this is a tool built for better information but not complete information
+        In addition, this system works on the Grade Distribution Releases located at https://www.clemson.edu/institutional-effectiveness/oir/data-reports/
+        *Course Sections that meet the following conditions are not included: Undergraduate classes with less than 10 students or Graduate classes with less than 5 students. In addition, if a section has all but 1 student making a common grade (example: All but one student makes a "B" in a class), the section is excluded.*
+        """
+        #Handle casing
+        prof_name = f'{firstName.lower().capitalize()} {lastName.lower().capitalize()}' 
+        
+        if prof_name not in self.global_master_prof_list:
+            embed = discord.Embed(title="Grades", color=Colors.Error)
+            result = 'That\'s not a professor at Clemson\n Are you sure you used the proper notation (ex: Brian Dean)?'
+            embed.add_field(name="ERROR: Professor doesn't exist", value=result, inline=False)
+            await ctx.send(embed=embed)
+        else:
```

Comment:
to reduce nesting we can put a return after line 368 then remove this else, turn it from an if else into an early exit condition which is the standard pattern in clembot. not a big deal just cuts down on nesting optionally